### ISSUE TITLE
Consider using XDG directory structure for cabal in CI

### ISSUE THREAD

**Issue by ulysses4ever** [haskell/cabal#8855]
As discussed in #8840, due to caching, our CI is probably stuck with ~/.cabal currently. It may be a good idea to wipe the cache and see where it brings us. Also, maybe it makes sense to add some diagnostics in the CI to make it more clear which setup we use (legacy or XDG).

**Comment by ulysses4ever** [comment]
Related:

- https://github.com/haskell/actions/issues/210

**Comment by andreasabel** [comment]
Can `cabal` tell me where it likes to store stuff?  Something like `stack path`.
```
$ stack path --stack-root
~/.stack
```

**Comment by mouse07410** [comment]
I **strongly** support this.

**Comment by mouse07410** [comment]
> Can `cabal` tell me where it likes to store stuff?  Something like `stack path`.
> 
> ```
> 
> $ stack path --stack-root
> 
> ~/.stack
> 
> ```


I **strongly** support this.

**Comment by ulysses4ever** [comment]
Well, there's a simple/ish algorithm written out in https://cabal.readthedocs.io/en/3.10/config.html#directories:

> Unless the CABAL_DIR environment variable is set or a ~/.cabal directory exists, Cabal will by default store data in directories according to the XDG Base Directory Specification. 
> ...
> `$XDG_STATE_HOME/cabal` for compiled libraries and other stateful artifacts, including the Cabal store.

But perhaps it would be handy to have such an interface to query cabal...

**Comment by andreasabel** [comment]
> But perhaps it would be handy to have such an interface to query cabal...

Yes, indeed, otherwise each tool around `cabal` has to duplicate this logic.
